mage_ai/io/qdrant.py

```diff
@@ -0,0 +1,151 @@
+                collection_name (str): name of the collection.
+                Defaults to the name defined in io_config.yaml.
+            Returns:
+                DataFrame: Data frame object loaded with data from qdrant
+        """
+        # Assume collection is already created and exists.
+        collection_name = collection_name or self.collection
+
+        hitted_results = self.client.search(
+            collection_name=collection_name,
+            query_vector=query_vector,
+            limit=limit_results,
+            with_vectors=True,
+        )
+        self.client.close()
+
+        output_df = {}
+        output_df['id'] = [hit.id for hit in hitted_results]
+        output_df['payload'] = [hit.payload for hit in hitted_results]
+        output_df['score'] = [hit.score for hit in hitted_results]
+        output_df['vector'] = [hit.vector for hit in hitted_results]
+
+        return DataFrame.from_dict(output_df)
+
+    def export(
+        self,
+        df: DataFrame,
+        document_column: str,
+        id_column: str = None,
+        vector_column: str = None,
+        collection_name: str = None,
+        vector_size: int = None,
+        distance: models.Distance = None,
+        **kwargs,
+    ) -> None:
+        """
+        Save data into Qdrant.
+        Args:
+            df (DataFrame): Data to export.
+            document_column (str): Column name of the document.
+            id_column (str): Column name of the id. Default will use index.
+            vector_column (str): Column name of the vector. Default will use encoder
+            to auto generate query vector.
+            collection_name (str): name of the collection.
+            vector_size (int): dimension size of vector.
+            distance (models.Distance): distance metric to use.
+        """
+        collection_name = collection_name or self.collection
+        encoder = SentenceTransformer("all-MiniLM-L6-v2")
+
+        try:
+            self.client.get_collection(collection_name)
+        except ValueError:
+            print(f'Creating collection: {collection_name}')
+            self.create_collection(
+                vector_size=vector_size or encoder.get_sentence_embedding_dimension(),
+                distance=distance,
+                collection_name=collection_name,
+            )
+
+        payloads = df[document_column].tolist()
+        if id_column is None:
+            ids = [x for x in df.index.tolist()]
+        else:
+            ids = df[id_column].tolist()
+        if vector_column is None:
+            vectors = [encoder.encode(str(x)).tolist() for x in payloads]
+        else:
+            vectors = df[vector_column].tolist()
+
+        self.client.upsert(
+            collection_name=collection_name,
+            points=models.Batch(
+                ids=ids,
+                payloads=payloads,
+                vectors=vectors,
+            ),
+        )
+        self.client.close()
```

Comment:
why do you close the client here? does that mean the export method can only be called once?
If you want to close the client when the class is deleted, you can add `__del__`  `__enter__` `__exit__` methods like https://github.com/mage-ai/mage-ai/blob/master/mage_ai/io/base.py#L360-L368

Reply:
Added `__del__`, `__enter__`, `__exit__`, `open` and `close` function to close or reopen the qdrant client.

Export function can be run multiple times successfully. I saw previously complaining`Storage folder /home/src/test7_qdrant.db is already accessed by another instance of Qdrant client. If you require concurrent access, use Qdrant server instead.` That's why add the close function at end.